### PR TITLE
Add development tutorial

### PR DESCRIPTION
**Pull request by zzxwill**
Moved development guides in README to DEVELOPMENT.md
and add PR to-do list.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+NAME                      AGE
+example-appconfig-trait   54s
+```
+
+## Prepare for Pull Request
+
+Before make a PR, please:
+
+1) write respective unit-test
+2) run `gofmt -s -w CHANGED_FILE` to format changed Go files
```
Comment (hongchaodeng) [inline DEVELOPMENT.md:79]: Take a look at https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md

Run `make reviweable` instead

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+1. Golang version 1.12+
+2. Kubernetes version v1.15+ with kubectl configured
+
+## Build
+
+The functionality of this library can be demonstrated with the following steps:
+
+* Fork and clone this project
+```shell script
+git clone https://github.com/xxxxxx/oam-kubernetes-runtime.git
```
Comment (wonderflow) [inline DEVELOPMENT.md:17]:
```suggestion
git clone git@github.com:crossplane/oam-kubernetes-runtime.git
```

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+## Prerequisites
+
+1. Golang version 1.12+
+2. Kubernetes version v1.15+ with kubectl configured
+
+## Build
+
+The functionality of this library can be demonstrated with the following steps:
+
+* Fork and clone this project
```
Comment (wonderflow) [inline DEVELOPMENT.md:15]:
```suggestion
* Clone this project
```

I think fork is required only if one guy want to contribute his code. We could write another doc or leave a github workflow doc link here.

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+```shell script
+git clone https://github.com/xxxxxx/oam-kubernetes-runtime.git
+```
+
+* Build the library 
+
+```shell
+make submodules 
+
+make
```
Comment (wonderflow) [inline DEVELOPMENT.md:25]: Maybe we can copy-paste the output together. Developers will be more confirmed if they have the same output.

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+## Make your change
+* Generate and install CRDs to your Kubernetes cluster
+
+```shell
+make generate
+
+kubectl apply -f crds/
+```
```
Comment (wonderflow) [inline DEVELOPMENT.md:35]: same here. copy-paste the output together

File: DEVELOPMENT.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+* Make your code changes
+
+Make changes and run the following to test it.
+
```
Comment (wonderflow) [inline DEVELOPMENT.md:40]: * `make reviewable` will generate and lint the code.
* `make test` will run tests.

Make sure these two steps are executed after every change.

File: DEVELOPMENT.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+If you only change code under [core apis](./apis/core), remember to
+regenerate crds mainifests as the step above.
+
+## Verification
```
Comment (wonderflow) [inline DEVELOPMENT.md:44]:
```suggestion
## Run a simple and basic workflow locally
```

File: DEVELOPMENT.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+You can start running OAM Kubernetes runtime to verify your chanes.
```
Comment (wonderflow) [inline DEVELOPMENT.md:45]: Before this, we need add some prerequisites for example: `minikube `

File: DEVELOPMENT.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+Before make a PR, please:
+
+1) write respective unit-test
+2) run `gofmt -s -w CHANGED_FILE` to format changed Go files
+3) run `make test` to run unit-tests
+
+## Clean up
+
+Delete all crds by `kubectl delete -f crds/` and other rouses you created.
```
Comment (wonderflow) [inline DEVELOPMENT.md:84]:
```suggestion
Delete all crds by `kubectl delete -f crds/` and other resources you created.
```

File: DEVELOPMENT.md (hunk 8 above)
Comment (wonderflow) [reply DEVELOPMENT.md:45]: I see you use ContainerizedWorkload as part of workflow, then it means user must install `https://github.com/crossplane/addon-oam-kubernetes-local` as prerequisite

File: DEVELOPMENT.md (hunk 1 above)
Comment (zzxwill) [reply DEVELOPMENT.md:79]: +1. Thanks.

File: DEVELOPMENT.md (hunk 8 above)
Comment (zzxwill) [reply DEVELOPMENT.md:45]: 1. Kubernetes cluster is a must in part `Prerequisites`. See https://github.com/crossplane/oam-kubernetes-runtime/pull/45/files#diff-f785e9e46b9afd2567df7a658338db17R9 please.
2. Apply manifests in crds will work. See https://github.com/crossplane/oam-kubernetes-runtime/pull/45/files#diff-f785e9e46b9afd2567df7a658338db17R34 please.

File: DEVELOPMENT.md (hunk 4 above)
Comment (zzxwill) [reply DEVELOPMENT.md:25]: +1

File: DEVELOPMENT.md (hunk 5 above)
Comment (zzxwill) [reply DEVELOPMENT.md:35]: +1

**Comment by hongchaodeng** [comment]
@zzxwill 
DCO did not pass. Please sign your commit

File: README.md [hunk 10]
```diff
@@ -13,60 +13,9 @@ In addition, we created this library with the following goals in mind
...
-example-appconfig-workload   12s
-```
-
-And a `Manualscalertrait` looking like below
-```
-kubectl get manualscalertraits.core.oam.dev
-NAME                      AGE
-example-appconfig-trait   54s
-```
+## To start developing OAM Kubernetes runtime
```
Comment (wonderflow) [inline README.md:17]:
```suggestion
## Getting started
```

File: README.md [hunk 11]
```diff
@@ -13,60 +13,9 @@ In addition, we created this library with the following goals in mind
...
+Check out [DEVELOPMENT.md](./DEVELOPMENT.md)
```
Comment (wonderflow) [inline README.md:18]:
```suggestion
Check out [DEVELOPMENT.md](./DEVELOPMENT.md) to see how to develop with OAM Kubernetes runtime
```

**Pull request merged**